java/core/src/main/java/com/whylogs/api/logger/resultSets/ResultSet.java

```diff
@@ -0,0 +1,41 @@
+package com.whylogs.api.logger.resultSets;
+
+import com.whylogs.core.DatasetProfile;
+import com.whylogs.core.errors.Error;
+import com.whylogs.core.metrics.Metric;
+import com.whylogs.core.views.DatasetProfileView;
+import java.util.Optional;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * A holder object for profiling results.
+ *
+ * <p>A whylogs.log call can result in more than one profile. This wrapper class simplifies the
+ * navigation among these profiles.
+ *
+ * <p>Note that currently we only hold one profile but we're planning to add other kinds of profiles
+ * such as segmented profiles here.
+ */
+@Data
+@NoArgsConstructor
+public abstract class ResultSet {
+
+  // TODO: implement read and write when I make the reader and writer
+
+  public abstract Optional<DatasetProfileView> view();
+
+  public abstract Optional<DatasetProfile> profile();
+
+  // TODO: Come back for ModelPerformanceMetrics
+
+  // Question: why is the python addMetrics when it only adds the one?
```

Comment:
Possibly a typo? It doesn't look like it's called anywhere in the code base, so rename it or maybe drop it?

Reply:
KK I'll update that then :)

Reply:
Keep it in sync with Python :)  (i.e., rename or drop in Python if we decide to go that way)